pkg/kubelet/kuberuntime/kuberuntime_container.go

```diff
@@ -809,6 +834,30 @@ func (m *kubeGenericRuntimeManager) RunInContainer(id kubecontainer.ContainerID,
 	return append(stdout, stderr...), err
 }
 
+// StartEphemeralContainer creates a Debug Container in a pod. It's called from the same goroutine as SyncPod()
+// TODO(verb) switch v1.Container to v1.EphemeralContainer when it is added to the API
+func (m *kubeGenericRuntimeManager) StartEphemeralContainer(pod *v1.Pod, podStatus *kubecontainer.PodStatus, container *v1.Container, pullSecrets []v1.Secret) error {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.DebugContainers) {
+		return errors.New("Debug Containers feature disabled")
+	}
+
+	if kubecontainer.GetContainerSpec(pod, container.Name) != nil {
```

Comment:
What if there's another ephemeral container with the same name? Is there a race condition here?